File: tools/abbrevs.js
var abbrevs = require('./abbrevs.json');
var abbrevsReplacements = [];

Object.keys(abbrevs).forEach(function(key) {
    for (var i = 0; i < abbrevs[key].length; i += 1) {
        abbrevsReplacements.push({
            'short': abbrevs[key][i],
            'long': key,
        });
    }
});

var parse = function parse(content) {
    content = content.replace(/([,\(] *)([A-Za-z ']*?)([,\)])/g, function(original, prefix, arg, postfix) {
        abbrevsReplacements.forEach(function(abbrev) {
            arg = arg.replace(RegExp('^' + abbrev.short + '$', 'i'), abbrev.long);
        });
        return prefix + arg + postfix;
    });
    
    return content;
}

exports.parse = parse;

File: tools/abbrevs.py
import json
import re
import os

# Load abbreviations from JSON file
script_dir = os.path.dirname(os.path.abspath(__file__))
with open(os.path.join(script_dir, 'abbrevs.json'), 'r') as f:
    abbrevs = json.load(f)

# Build abbreviation replacements
abbrevs_replacements = []

for key in abbrevs:
    for short in abbrevs[key]:
        abbrevs_replacements.append({
            'short': short,
            'long': key,
        })

def expand(abbrev):
    """Expand a single abbreviation to its full form"""
    for a in abbrevs_replacements:
        abbrev = re.sub(r'^' + a['short'] + r'$', a['long'], abbrev, flags=re.IGNORECASE)
    return abbrev

def parse(content):
    """Parse content and expand abbreviations in function arguments"""
    def replace_abbrev(match):
        prefix = match.group(1)
        arg = match.group(2)
        postfix = match.group(3)
        arg = expand(arg)
        return prefix + arg + postfix
    
    content = re.sub(r"([,\(] *)([A-Za-z ']*?)([,\)])", replace_abbrev, content)
    return content
